src/tools/pub/publishing.md

```diff
@@ -268,6 +268,46 @@ the package page displays links to both the prerelease and the stable release.
 The prerelease doesn't affect the analysis score, show up in search results,
 or replace the package `README.md` and documentation.
 
+## Publishing previews
+
+Previews can be useful when **all** of the following are true:
+
+* Your next stable version of the package is complete.
+
+* You depend on an API or feature from the Dart SDK which hasn't yet been
+  released in a stable Dart SDK.
+
+* You know that the API or feature you depend on is API stable, and will not be
+  changing prior to reaching stable.
+
+A concrete example is a `package:args` which has a finished version `2.0.0` but
+which depends on a feature in Dart `2.12.0-259.8.beta`, and where Dart SDK
+version `2.12.0` stable hasn't released yet. The pubspec for such a package
+would look like this:
+
+```
+name: args
+version: 2.0.0
+environment:
+  sdk: '>=2.12.0-259.8.beta <3.0.0'
+```
+
+When publishing such a package to pub.dev, it will be tagged as a Preview
+version as illustated by this screenshow where the stable version is listed as
+`1.6.0` and the preview version is listed as `2.0.0.`.
+
+![Illustration of a preview version](/null-safety/preview-version.png){:width="600px"}<br>
+
+When a Dart `2.12.0` stable release is finished, and deployed to pub.dev, the
+package will automatically be listed to display `2.0.0` as the stable version.
+
+Note that when running `pub publish` you will see this warning. You can ignore
```

Comment:
```suggestion
Note that when running `dart pub publish` you will see this warning. You can ignore
```

Reply:
Oops, didn't see this before making my changes. I've copied it below.